def fun1():
    print(x)
fun1()
x=100

# it gives error as fun1() is called before defining x value
#NameError: name 'x' is not defined
